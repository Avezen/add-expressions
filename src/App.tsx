import React, {useEffect, useState} from 'react';
import './App.css';
import {algebra} from "./util/algebra";
import {Expression} from "./component/Expression/Expression";
import {ExpressionsSum} from "./component/Expression/ExpressionsSum";

const App = () => {
    const [firstExpression, setFirstExpression] = useState([{factor: 5, power: 2}]);
    const [secondExpression, setSecondExpression] = useState([{factor: 2, power: 0}]);
    const [sumOfExpressions, setSumOfExpressions] = useState({});

    useEffect(() => {
        setSumOfExpressions(algebra.sumExpressions(firstExpression, secondExpression));
    }, [firstExpression, secondExpression]);

    return (
        <div className="App">
            <br/>
            First expression:
            <Expression
                expression={firstExpression}
                setExpression={setFirstExpression}
            />
            <br/>
            Second expression:
            <Expression
                expression={secondExpression}
                setExpression={setSecondExpression}
            />
            <br/>
            Result:
            <ExpressionsSum
                sumOfExpressions={sumOfExpressions}
            />
        </div>
    );
};

export default App;
