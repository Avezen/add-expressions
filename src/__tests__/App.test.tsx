import React from "react";
import renderer from 'react-test-renderer';
import App from "../App";


const renderApp = () => {
    return <App/>;
};

describe("<App />", () => {
    it("should render component", () => {
        const findByTestId = renderApp(),
            ExpressionsSumComponent = renderer.create(findByTestId).toJSON();

        expect(ExpressionsSumComponent).toMatchSnapshot();
    });

    it("should render proper result (5x2 + 2)", () => {
        const findByTestId = renderApp(),
            ExpressionsSumComponent = renderer.create(findByTestId).toJSON();

        expect(ExpressionsSumComponent).toMatchSnapshot();
    });
});
