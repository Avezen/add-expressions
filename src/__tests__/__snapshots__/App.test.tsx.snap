// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render component 1`] = `
<div
  className="App"
>
  <br />
  First expression:
  <div>
    <button
      id="add-expression"
      onClick={[Function]}
    >
      +
    </button>
    <br />
    <div
      className="expression"
    >
      <div>
        <input
          id="factor-0"
          onChange={[Function]}
          type="number"
          value={5}
        />
        x
        <input
          id="power-0"
          onChange={[Function]}
          type="number"
          value={2}
        />
        <button
          id="remove-expression-0"
          onClick={[Function]}
        >
          x
        </button>
        <br />
      </div>
    </div>
  </div>
  <br />
  Second expression:
  <div>
    <button
      id="add-expression"
      onClick={[Function]}
    >
      +
    </button>
    <br />
    <div
      className="expression"
    >
      <div>
        <input
          id="factor-0"
          onChange={[Function]}
          type="number"
          value={2}
        />
        x
        <input
          id="power-0"
          onChange={[Function]}
          type="number"
          value={0}
        />
        <button
          id="remove-expression-0"
          onClick={[Function]}
        >
          x
        </button>
        <br />
      </div>
    </div>
  </div>
  <br />
  Result:
  <div
    className="expression"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  />
</div>
`;

exports[`<App /> should render proper result (5x2 + 2 1`] = `
<div
  className="App"
>
  <br />
  First expression:
  <div>
    <button
      id="add-expression"
      onClick={[Function]}
    >
      +
    </button>
    <br />
    <div
      className="expression"
    >
      <div>
        <input
          id="factor-0"
          onChange={[Function]}
          type="number"
          value={5}
        />
        x
        <input
          id="power-0"
          onChange={[Function]}
          type="number"
          value={2}
        />
        <button
          id="remove-expression-0"
          onClick={[Function]}
        >
          x
        </button>
        <br />
      </div>
    </div>
  </div>
  <br />
  Second expression:
  <div>
    <button
      id="add-expression"
      onClick={[Function]}
    >
      +
    </button>
    <br />
    <div
      className="expression"
    >
      <div>
        <input
          id="factor-0"
          onChange={[Function]}
          type="number"
          value={2}
        />
        x
        <input
          id="power-0"
          onChange={[Function]}
          type="number"
          value={0}
        />
        <button
          id="remove-expression-0"
          onClick={[Function]}
        >
          x
        </button>
        <br />
      </div>
    </div>
  </div>
  <br />
  Result:
  <div
    className="expression"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <span>
      +
    </span>
    5
    x
    <sup>
      <small>
        2
      </small>
    </sup>
    <span>
      +
    </span>
    2
  </div>
</div>
`;
